File: 0153-find-minimum-in-rotated-sorted-array/0153-find-minimum-in-rotated-sorted-array.py
class Solution:
    def findMin(self, nums: List[int]) -> int:
        min_val = nums[0]
        left = 0
        right = len(nums)-1
        
        while left <= right:
            mid = left + (right-left)//2
            
            if nums[left] <= nums[mid]:
                min_val = min(min_val,nums[left])
                left = mid + 1
            else:
                min_val = min(min_val,nums[mid])
                right = mid - 1
                
        return min_val
                

File: 0153-find-minimum-in-rotated-sorted-array/0153-find-minimum-in-rotated-sorted-array.java
class Solution {
    public int findMin(int[] nums) {
        int left = 0;
        int right = nums.length - 1;
        int mini = nums[0];

        while (left <= right){
            int mid = left + (right - left) / 2;

            if (nums[left] <= nums[mid]){
                mini = Math.min(mini, nums[left]);
                left = mid + 1;
            } else{
                mini = Math.min(mini, nums[mid]);
                right = mid - 1;
            }
        }

        return mini;
    }
}